raphtory-graphql/src/model/graph/timeindex.rs

```diff
@@ -0,0 +1,128 @@
+use async_graphql::{Error, Value as GqlValue};
+use dynamic_graphql::{
+    InputObject, OneOfInput, ResolvedObject, ResolvedObjectFields, Scalar, ScalarValue,
+};
+use raphtory_api::core::{
+    storage::timeindex::{AsTime, TimeError, TimeIndexEntry},
+    utils::time::{IntoTime, ParseTimeError, TryIntoTime},
+};
+
+/// Input for primary time component. Int or DateTime formatted String.
+/// Valid formats are RFC3339, RFC2822, %Y-%m-%d, %Y-%m-%dT%H:%M:%S%.3f, %Y-%m-%dT%H:%M:%S%,
+/// %Y-%m-%d %H:%M:%S%.3f and %Y-%m-%d %H:%M:%S%.
+#[derive(Scalar, Clone, Debug)]
+#[graphql(name = "SimpleTimeInput")]
+pub struct GqlSimpleTimeInput(pub i64);
+
+impl ScalarValue for GqlSimpleTimeInput {
+    fn from_value(value: GqlValue) -> Result<Self, Error> {
+        match value {
+            GqlValue::Number(timestamp) => timestamp
+                .as_i64()
+                .ok_or(Error::new("Expected Int or DateTime formatted String."))
+                .map(|timestamp| GqlSimpleTimeInput(timestamp)),
+            GqlValue::String(dt) => dt
+                .try_into_time()
+                .map(|t| GqlSimpleTimeInput(t.t()))
+                .map_err(|e| Error::new(e.to_string())),
+            _ => Err(Error::new("Expected Int or DateTime formatted String.")),
```

Comment:
If we allow object here, can we get away without the OneOfInput later which forces us to have named fields, i.e., can we just accept things like `Number`, `String`, `{epoch: Number | String, index: Number}`?